src/topology/algebra/infinite_sum/basic.lean

```diff
@@ -313,6 +318,16 @@ lemma has_sum.compl_add {s : set β} (ha : has_sum (f ∘ coe : sᶜ → α) a)
   has_sum f (a + b) :=
 ha.add_is_compl is_compl_compl.symm hb
 
+lemma has_sum.sum_t {f : β ⊕ γ → α} {a_inl a_inr : α} (h_inl : has_sum (f ∘ sum.inl) a_inl)
```

Comment:
There are two ways to state this lemma: starting with `β ⊕ γ → α` and starting with two functions and combining them using `sum.elim`. Please add the second version too (the proof should be by application of this version). I'm not sure about the naming convention for this kind of lemmas. I would ask on Zulip.